core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -647,6 +647,13 @@ private[spark] object JsonProtocol {
     val taskInfo = taskInfoFromJson(json \ "Task Info")
     val executorMetrics = executorMetricsFromJson(json \ "Task Executor Metrics")
     val taskMetrics = taskMetricsFromJson(json \ "Task Metrics")
+    val peakExecutionMemory = taskInfo.accumulables.find(accInfo => {
```

Comment:
This code should be in `taskMetricsFromJson` like other metrics, following the style over there.

Also, please add a unit test.

Reply:
Thanks for your reply, I will add soon

Reply:
hi, @vanzin , I update this PR

the peakExecutionMemory value zero happend when replaying event log, so I add the metrics value in SparkListenerTaskEnd json info and parsing it when replaying and set the value back like other metrics do, this is a litter more natural than the first commit for this PR

the unit test is the same as JsonProtocolSuite.testTaskMetrics 

could you please kindly review?

Reply:
hi, @vanzin , I update this PR

the peakExecutionMemory value zero happend when replaying event log, so I add the metrics value in SparkListenerTaskEnd json info and parsing it when replaying and set the value back like other metrics do, this is a litter more natural than the first commit for this PR

the unit test is the same as JsonProtocolSuite.testTaskMetrics

could you please kindly review?

Reply:
You need to update that unit tests to verify the new value you're adding.

Reply:
thanks, I update that unit test and take a successful test before push.